src/frequenz/core/asyncio/_task_group.py

```diff
@@ -0,0 +1,392 @@
+    """
+
+    def __init__(
+        self, *, unique_id: str | None = None, task_creator: TaskCreator = asyncio
+    ) -> None:
+        """Initialize this instance.
+
+        Args:
+            unique_id: The string to uniquely identify this instance. If `None`,
+                a string based on `hex(id(self))` will be used. This is used in
+                `__repr__` and `__str__` methods, mainly for debugging purposes, to
+                identify a particular instance of a persistent task group.
+            task_creator: The object that will be used to create tasks. Usually one of:
+                the [`asyncio`]() module, an [`asyncio.AbstractEventLoop`]() or
+                an [`asyncio.TaskGroup`]().
+        """
+        # [2:] is used to remove the '0x' prefix from the hex representation of the id,
+        # as it doesn't add any uniqueness to the string.
+        self._unique_id: str = hex(id(self))[2:] if unique_id is None else unique_id
+        """The unique ID of this instance."""
+
+        self._task_creator: TaskCreator = task_creator
+        """The object that will be used to create tasks."""
+
+        self._running: set[asyncio.Task[Any]] = set()
+        """The set of tasks that are still running.
+
+        Tasks are removed from this set automatically when they finish using the
+        Task.add_done_callback method.
+        """
+
+        self._waiting_ack: set[asyncio.Task[Any]] = set()
+        """The set of tasks that have finished but waiting for the user's ACK.
+
+        Tasks are added to this set automatically when they finish using the
+        Task.add_done_callback method.
+        """
+
+    @property
+    def unique_id(self) -> str:
+        """The unique ID of this instance."""
+        return self._unique_id
+
+    @property
+    def tasks(self) -> Set[asyncio.Task[Any]]:
+        """The set of tasks managed by this group.
+
+        Users typically should not modify the tasks in the returned set and only use
+        them for informational purposes.
+
+        Both running tasks and tasks pending for acknowledgment are included in the
+        returned set.
+
+        Danger:
+            Changing the returned tasks may lead to unexpected behavior, don't do it
+            unless the class explicitly documents it is safe to do so.
+        """
+        return self._running | self._waiting_ack
+
+    @property
+    def task_creator(self) -> TaskCreator:
+        """The object that will be used to create tasks."""
+        return self._task_creator
+
+    @property
+    def is_running(self) -> bool:
+        """Whether this task group is running.
+
+        A task group is considered running when at least one task is running.
+        """
+        return bool(self._running)
+
+    def create_task(
+        self,
+        coro: Coroutine[Any, Any, TaskReturnT],
+        *,
+        name: str | None = None,
+        context: contextvars.Context | None = None,
+        log_exception: bool = True,
```

Comment:
I'm not so sure about this, if it is easy to check for completed tasks to handle errors, maybe we don't need the logging feature (or we'll get most code littered with `log_exceptions=False`). But if anyone forgets, then we are screwed again with task dying without notice.

I would love to find a way to log only if the task was not ACKed somehow, but I couldn't so far.

I guess leaving it for now is the safest bet, if people need to opt-out explicitly at least there are much bigger chances that they will handle the exceptions. And for a casual user that doesn't know how to do it right, at least we have a log message.